### ISSUE TITLE
Code in guide "Scripting with Quarkus" does not run with jbang

### ISSUE THREAD

**Issue by cdhermann** [quarkusio/quarkus#24926]
### Describe the bug

The code https://quarkus.io/guides/scripting#solution can't be run with `jbang`. A stracktrace is shown. As soon as the dependency `io.quarkus:quarkus-resteasy-reactive:2.8.0.Final` is replaced by `io.quarkus:quarkus-resteasy:2.8.0.Final` the script runs.

### Expected behavior

Application startup without a stacktrace showing the Quarkus banner and the installed features. Running `curl http://localhost:8080/hello` prints `hello`

### Actual behavior

The following stacktrace is shown

```
Exception in thread "main" java.lang.NoClassDefFoundError: org/eclipse/microprofile/config/ConfigValue
	at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:449)
	at io.quarkus.runtime.generated.Config.<clinit>(Unknown Source)
	at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.ReflectAccess.newInstance(ReflectAccess.java:128)
	at java.base/jdk.internal.reflect.ReflectionFactory.newInstance(ReflectionFactory.java:347)
	at java.base/java.lang.Class.newInstance(Class.java:645)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:41)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:120)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Caused by: java.lang.ClassNotFoundException: org.eclipse.microprofile.config.ConfigValue
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	... 14 more
```

### How to Reproduce?

1. install JBang; e.g. `curl -Ls https://sh.jbang.dev | bash -s - app setup`
2. save the code from https://quarkus.io/guides/scripting#solution in a file called `quarkusapp.java`
3. run the code using `jbang run quarkusapp.java`

### Output of `uname -a` or `ver`

Darwin Kernel Version 21.4.0

### Output of `java -version`

ava version "17" 2021-09-14 LTS Java(TM) SE Runtime Environment (build 17+35-LTS-2724) Java HotSpot(TM) 64-Bit Server VM (build 17+35-LTS-2724, mixed mode, sharing)

### Quarkus version or git rev

2.8.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

jbang version 0.92.2

**Comment by geoand** [comment]
Seems to be some kind of dependency version problem...
Anyway in JBang to see the dependency tree that was used?

**Comment by cdhermann** [comment]
@geoand I'm not quite sure if this is the right way to produce a dependency tree but my approach was

1. start in editing mode with `jbang edit --open=code --live quarkusapp.java`
2. open a terminal inside the editor / IDE
3. run `gradle -q dependencies > deps.txt`

Probably every other editor / IDE will do it.

[The deps.txt file produced is attached](https://github.com/quarkusio/quarkus/files/8487407/deps.txt).

**Comment by geoand** [comment]
My question was more towards @maxandersen who is the creator of jbang :)

**Comment by maxandersen** [comment]
Run with jbang --verbose and it will show classpath. 

'jbang info tools app.java' should also list it.

**Comment by maxandersen** [comment]
Btw. The issue is probably that the example does not use the bom (we should update it to use that) but it's weird that reactive resteasy as the top dependency does not work out of the box. 

That indicates something wrong/weird in its default dependency chain.

**Comment by geoand** [comment]
> Btw. The issue is probably that the example does not use the bom (we should update it to use that) but it's weird that reactive resteasy as the top dependency does not work out of the box.

Very good point! Adding the BOM fixes the issue. So I'll update the docs to include the BOM

**Comment by maxandersen** [comment]
btw. i'm on a really crappy 4g connection in rural germany autobahn somewhere and somehow 2.7.x resolves much faster than 2.8.x dependencies. just an observation i find weird.

**Comment by geoand** [comment]
https://github.com/quarkusio/quarkus/pull/24933 is the PR for the docs

**Comment by geoand** [comment]
> btw. i'm on a really crappy 4g connection in rural germany autobahn somewhere and somehow 2.7.x resolves much faster than 2.8.x dependencies. just an observation i find weird.

no idea what the problem might be

**Comment by cdhermann** [comment]
unfortunately off-topic but related:
@maxandersen is there also an option to just show the dependency tree without revealing full paths on the system?

**Comment by cdhermann** [comment]
> btw. i'm on a really crappy 4g connection in rural germany autobahn somewhere and somehow 2.7.x resolves much faster than 2.8.x dependencies. just an observation i find weird.

2.8.0.Finall additionally pulls in `smallrye-common-vertx-context-1.10.0`. But this is only about 5,2kB

**Comment by maxandersen** [comment]
@cdhermann you can open issue at jbangdev/jbang on getting a GAV list
